src/zenml/zen_stores/schemas/logs_schemas.py

```diff
@@ -0,0 +1,88 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""SQLModel implementation of pipeline logs tables."""
+
+
+from typing import Optional
+from uuid import UUID
+
+from sqlalchemy import TEXT, Column
+from sqlmodel import Field, Relationship
+
+from zenml.models.logs_models import LogsResponseModel
+from zenml.zen_stores.schemas.base_schemas import BaseSchema
+from zenml.zen_stores.schemas.component_schemas import StackComponentSchema
+from zenml.zen_stores.schemas.pipeline_run_schemas import PipelineRunSchema
+from zenml.zen_stores.schemas.schema_utils import build_foreign_key_field
+from zenml.zen_stores.schemas.step_run_schemas import StepRunSchema
+
+
+class LogsSchema(BaseSchema, table=True):
+    """SQL Model for logs."""
+
+    __tablename__ = "logs"
+
+    pipeline_run_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=PipelineRunSchema.__tablename__,
+        source_column="pipeline_run_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=True,
+    )
+    pipeline_run: Optional["PipelineRunSchema"] = Relationship(
+        back_populates="run_logs"
+    )
+
+    step_run_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=StepRunSchema.__tablename__,
+        source_column="step_run_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=True,
+    )
+    step_run: Optional["StepRunSchema"] = Relationship(
+        back_populates="step_logs"
+    )
+
+    artifact_store_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=StackComponentSchema.__tablename__,
+        source_column="stack_component_id",
+        target_column="id",
+        ondelete="SET NULL",
```

Comment:
This is inconsistent with the relationship defines in `StackComponentSchema` which defines `cascade` as delete behavior.

If the artifact store is deleted we cannot read the logs anymore, so cascade makes sense IMO.

```suggestion
        ondelete="CASCADE",
```